Source/UnitEx.cpp

```diff
@@ -1144,7 +1144,7 @@ int UnitEx::GetNumTruckLoadsSoFar()
 		return HFLNOTINITED;
 	}
 
-	BeaconData* p = (BeaconData*)((int)(*unitArray) + (unitID * sizeof(OP2Unit)) + beaconDataOffset);
+	BeaconData* p = (BeaconData*)(reinterpret_cast<std::uintptr_t>(*unitArray) + (unitID * sizeof(OP2Unit)) + beaconDataOffset);

```

Comment:
Minor point: The `(BeaconData*)` is also a C-style cast, which should probably be updated.

Reply:
Will be fixed shortly